src/coreclr/vm/pefile.cpp

```diff
@@ -2280,19 +2274,10 @@ void PEFile::EnsureImageOpened()
 
 void PEFile::SetupAssemblyLoadContext()
 {
-    PTR_ICLRPrivBinder pBindingContext = GetBindingContext();
-    ICLRPrivBinder* pOpaqueBinder = NULL;
-
-    if (pBindingContext != NULL)
-    {
-        UINT_PTR assemblyBinderID = 0;
-        IfFailThrow(pBindingContext->GetBinderID(&assemblyBinderID));
+    PTR_AssemblyBinder pBindingContext = GetBindingContext();
 
-        pOpaqueBinder = reinterpret_cast<ICLRPrivBinder*>(assemblyBinderID);
-    }
-
-    m_pAssemblyLoadContext = (pOpaqueBinder != NULL) ?
-        (AssemblyLoadContext*)pOpaqueBinder :
+    m_pAssemblyLoadContext = (pBindingContext != NULL) ?
+        (AssemblyLoadContext*)pBindingContext :
```

Comment:
Just curious: This means that all `AssemblyBinder` instances are also `AssemblyLoadContext` instances. Maybe in the future we can remove another layer of abstraction...

Reply:
I thought about merging AssemblyBinder and AssemblyLoadContext an decided to keep them separate on purpose.  

AssemblyLoadContext should keep providing the link to the managed ALC, but should itself not be involved in binding business.

Reply:
Well - it has to be involved by the virtue of having a link to the managed extensions points (events and so on). But I guess it's OK to view them as slightly separate.

Reply:
At the first glance I saw enough usage of `AssemblyLoadContext` on the native side, but looking closer it mostly refers to the the managed one or to its name. The part that it matches the name of the managed type now may not be all that helpful.

Perhaps we do not need AssemblyLoadContext on the native side and should just move its stuff to the `AssemblyBinder`.

I need to think about this. 
It is easier to merge types than to un-merge them back later, if we regret it :-)

Reply:
> It is easier to merge types than to un-merge them back later, if we regret it :-)

That is very true. 👍